garb/garb_main.cpp

```diff
@@ -13,6 +13,80 @@
 #include <thread>
 #include <chrono>
 
+#if defined(WITH_COREDUMPER) && WITH_COREDUMPER
+#include "coredumper/coredumper.h"
+
+#include <signal.h>
+
+void my_write_libcoredumper(int sig, const char *path, time_t curr_time) {
+  int ret = 0;
+  static constexpr std::size_t buf_size = 512;
+  char suffix[buf_size];
+  char core[buf_size];
+  memset(suffix, '\0', buf_size);
+  memset(core, '\0', buf_size);
+  struct tm *timeinfo = gmtime(&curr_time);
+
+  if (path == nullptr)
+    strcpy(core, "core");
+  else
+    strncpy(core, path, buf_size - 1);
+
+  sprintf(suffix, ".%d%02d%02d%02d%02d%02d", (1900 + timeinfo->tm_year),
+      timeinfo->tm_mon, timeinfo->tm_mday, timeinfo->tm_hour,
+      timeinfo->tm_min, timeinfo->tm_sec);
+  strncat(core, suffix, buf_size - strlen(core) - 1);
+  static constexpr auto core_msg = "CORE PATH: ";
+  write(STDERR_FILENO, core_msg, strlen(core_msg));
+  write(STDERR_FILENO, core, strlen(core));
+  write(STDERR_FILENO, "\n\n", 2);
+  ret = WriteCoreDump(core);
+  if (ret != 0) {
+    static constexpr auto err_msg = "Error writing coredump.";
+    write(STDERR_FILENO, err_msg, strlen(err_msg));
+  }
+}
+
+std::string coredumper_core_path;
+
+extern "C" void handle_fatal_signal(int sig) {
+  my_write_libcoredumper(sig, coredumper_core_path.c_str(), time(nullptr));
```

Comment:
Not a strong opinion here, but what is the advantage of passing the result of time() via parameter?
It could be done inside my_write_libcoredumper(), right?